src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
   EXPECT_FALSE(unit_test_vertical_slider->IsMouseOver());
 }
 
+class CaptureClientAppInterfaceFake : public orbit_capture_client::AppInterface {
+  [[nodiscard]] orbit_capture_client::CaptureClient::State GetCaptureState() const {
+    return orbit_capture_client::CaptureClient::State::kStopped;
+  }
+
+  [[nodiscard]] bool IsCapturing() const { return false; }
+
+  void StartCapture(){};
+  void StopCapture(){};
+  void AbortCapture(){};
+  void ToggleCapture(){};
+};
+
+class NavigationTestCaptureWindow : public CaptureWindow, public testing::Test {
+ public:
+  explicit NavigationTestCaptureWindow() : CaptureWindow(nullptr) {
+    capture_client_app_ = &capture_client_app_fake_;
+
+    Resize(600, 100);
+
+    capture_data_ = TrackTestData::GenerateTestCaptureData();
+    time_graph_ = std::make_unique<TimeGraph>(this, nullptr, &viewport_, capture_data_.get(),
+                                              &picking_manager_);
+
+    CreateSchedulerTimers();
+
+    // Prepare the TimeGraph to be fully zoomed out
+    UpdateChildrenPosAndSize();
+    time_graph_->ZoomAll();
+    time_graph_->UpdateLayout();
+    UpdateChildrenPosAndSize();
+
+    // Make sure our expectations about the height of the timeline are correct,
+    // otherwise zooming may not work as expected - if this fails, the overall
+    // viewport height should be changed and the test adjusted accordingly
+    assert(time_graph_->GetTimelineUi()->GetHeight() < 95);
```

Comment:
Nit/personal preference: Those are non-test assertions, right? I would use ORBIT_CHECK in that case.

Reply:
Both done